sunpy/map/mapbase.py

```diff
@@ -1706,11 +1706,14 @@ def rotate(self, angle: u.deg = None, rmatrix=None, order=3, scale=1.0,
         if version.parse(np.__version__) < version.parse("1.20.0") and issubclass(self.data.dtype.type, numbers.Integral) and (missing % 1 != 0):
             warn_user("The specified `missing` value is not an integer, but the data "
                       "array is of integer type, so the output may be strange.")
-        # Pad the image array
-        new_data = np.pad(self.data,
-                          ((pad_y, pad_y), (pad_x, pad_x)),
-                          mode='constant',
-                          constant_values=(missing, missing))
+        try:
+            new_data = np.pad(self.data, 
+                      ((pad_y, pad_y), (pad_x, pad_x)), 
+                      mode='constant', 
+                      constant_values=(missing, missing))
+        except ValueError :
+            error_message = f"Error occurred during padding as constant values are NaN(not a number)"
+            warn_user(error_message)
```

Comment:
Can you add a unit test to ensure we hit this error please?

Reply:
do you want me to add unit test in this code?